lib/tests/test_sweeps.c

```diff
@@ -370,6 +370,61 @@ sweep_genic_selection_mimic_msms_single_run(unsigned long int seed)
     tsk_table_collection_free(&tables);
 }
 
+static void
+sweep_genic_selection_reverse_single_run(unsigned long int seed)
+{
+
+    /* Try to mimic the msms parameters used in verification.py
+           "100 300 -t 200 -r 200 500000"
+           " -SF 0 0.9 -Sp 0.5 -SaA 5000 -SAA 10000 -N 10000"
+     */
+    int ret;
+    const char *filename
+        = "/home/aalhadbhatt/Desktop/msprime/notebooks/PyNBmsprime/events.bin";
+    uint32_t n = 10;
+    double num_loci = 500001;
+    double position = num_loci / 2;
+    size_t num_demes = 1;
+    double recom_rate = 0.0004;
+    msp_t msp;
+    gsl_rng *rng = safe_rng_alloc();
+    tsk_table_collection_t tables;
+
+    // Test over differnt seeds
+    gsl_rng_set(rng, seed);
+
+    ret = build_sim(&msp, &tables, rng, num_loci, num_demes, NULL, n);
+    CU_ASSERT_EQUAL(ret, 0);
+    CU_ASSERT_EQUAL_FATAL(msp_set_recombination_rate(&msp, recom_rate), 0);
+    ret = msp_set_num_labels(&msp, 2);
+    CU_ASSERT_EQUAL(ret, 0);
+
+    // To mimic the verfication.py call
+    msp_set_discrete_genome(&msp, 0);
+    msp_set_gene_conversion_rate(&msp, 0);
+    msp_set_gene_conversion_tract_length(&msp, 1);
+    msp_set_avl_node_block_size(&msp, 65536);
+    msp_set_node_mapping_block_size(&msp, 65536);
+    msp_set_segment_block_size(&msp, 65536);
+
+    ret = msp_set_simulation_model_sweep_genic_selection_reverse(
```

Comment:
So, here instead of passing in the filename we would pass parameters required to describe the sweep.

Reply:
For the C code tests we won't need anything realistic, just some way of generating trajectories that'll test all the different code paths.